src/logger/log_entry.py

```diff
@@ -0,0 +1,166 @@
+
+        train_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainSpawnLogEntry":
+            return TrainSpawnLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+
+
+class TrainRemoveLogEntry(LogEntry):
+    """A LogEntry that represents the removal of a train."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainRemoveLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainRemoveLogEntry":
+            return TrainRemoveLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+
+
+class TrainArrivalLogEntry(LogEntry):
+    """A LogEntry that represents the arrival of a train at a station."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainArrivalLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
+        station_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainArrivalLogEntry":
+            return TrainArrivalLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+    station_id = IntegerField(null=False)
+
+
+class TrainDepartureLogEntry(LogEntry):
+    """A LogEntry that represents the departure of a train from a station."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the TrainDepartureLogEntry model."""
+
+        train_id = marsh.fields.Integer(required=True)
+        station_id = marsh.fields.Integer(required=True)
+
+        def _make(self, data: dict) -> "TrainDepartureLogEntry":
+            return TrainDepartureLogEntry(**data)
+
+    train_id = IntegerField(null=False)
+    station_id = IntegerField(null=False)
+
+
+class CreateFahrstrasseLogEntry(LogEntry):
+    """A LogEntry that represents the creation of a fahrstrasse."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the CreateFahrstrasseLogEntry model."""
+
+        fahrstrasse = marsh.fields.String(required=True)
+
+        def _make(self, data: dict) -> "CreateFahrstrasseLogEntry":
+            return CreateFahrstrasseLogEntry(**data)
+
+    fahrstrasse = CharField(null=False)
+
+
+class RemoveFahrstrasseLogEntry(LogEntry):
+    """A LogEntry that represents the removal of a fahrstrasse."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the RemoveFahrstrasseLogEntry model."""
+
+        fahrstrasse = marsh.fields.String(required=True)
+
+        def _make(self, data: dict) -> "RemoveFahrstrasseLogEntry":
+            return RemoveFahrstrasseLogEntry(**data)
+
+    fahrstrasse = CharField(null=False)
+
+
+class SetSignalLogEntry(LogEntry):
+    """A LogEntry that represents the setting of a signal."""
+
+    class Schema(LogEntry.Schema):
+        """The marshmallow schema for the SetSignalLogEntry model."""
+
+        signal_id = marsh.fields.String(required=True)
```

Comment:
Please discuss with @SaturnHafen or @Lietze  if the id of a signal is an integer, uuid or string 😇 I'm unsure at the moment.
(Please update it in the other classes too 🥳)

Reply:
Will change it to UUID